src/Umbraco.Infrastructure/Persistence/UmbracoDatabase.cs

```diff
@@ -92,6 +92,43 @@ private void Init()
             {
                 Mappers.AddRange(_mapperCollection);
             }
+
+            InitCommandTimeout();
+        }
+
+        // https://github.com/umbraco/Umbraco-CMS/issues/13354
+        // This sets the Database Command to connectionString Connection Timeout /  Connect Timeout
+        // This could be better, ideally the UmbracoDatabaseFactory.CreateDatabase() function would set this based on a setting (global or connectionstring setting)
+        private void InitCommandTimeout()
+        {
+            if (this.Connection != null && this.Connection.ConnectionTimeout > 0)
+            {
+                this.CommandTimeout = this.Connection.ConnectionTimeout;
+            }
+            else // get from _connectionString
+            {
+                var connectionParser = new DbConnectionStringBuilder
+                {
+                    ConnectionString = ConnectionString
+                };
+
+                if (connectionParser.ContainsKey("connection timeout"))
+                {
+                    if (int.TryParse(connectionParser["connection timeout"].ToString(), out int connectionTimeout))
+                    {
+                        _logger.LogInformation($"Setting Command Timeout to value configured in connectionstring Connection Timeout : {connectionTimeout} seconds");
+                        this.CommandTimeout = connectionTimeout;
+                    }
+                }
+                else if (connectionParser.ContainsKey("connect timeout"))
+                {
+                    if (int.TryParse(connectionParser["connect timeout"].ToString(), out int connectionTimeout))
+                    {
+                        _logger.LogInformation($"Setting Command Timeout to value configured in connectionstring Connect Timeout : {connectionTimeout} seconds");
+                        this.CommandTimeout = connectionTimeout;
+                    }
+                }
+            }
         }
```

Comment:
```suggestion
        private void InitCommandTimeout()
        {
            if (CommandTimeout != 0)
            {
                // CommandTimeout configured elsewhere, so we'll skip
                return;
            }

            if (Connection is not null && Connection.ConnectionTimeout > 0)
            {
                CommandTimeout = Connection.ConnectionTimeout;
                return;
            }

            // get from ConnectionString
            var connectionParser = new DbConnectionStringBuilder
            {
                ConnectionString = ConnectionString
            };

            if (connectionParser.TryGetValue("connection timeout", out var connectionTimeoutString))
            {
                if (int.TryParse(connectionTimeoutString.ToString(), out var connectionTimeout))
                {
                    _logger.LogTrace("Setting Command Timeout to value configured in connectionstring Connection Timeout : {TimeOut} seconds", connectionTimeout);
                    CommandTimeout = connectionTimeout;
                    return;
                }
            }

            if (connectionParser.TryGetValue("connect timeout", out var connectTimeoutString))
            {
                if (int.TryParse(connectTimeoutString.ToString(), out var connectionTimeout))
                {
                    _logger.LogTrace("Setting Command Timeout to value configured in connectionstring Connect Timeout : {TimeOut} seconds", connectionTimeout);
                    CommandTimeout = connectionTimeout;
                }
            }
        }
```

I figured it might be a good idea to check if `CommandTimeout` is already configured, and then skip this logic if it is 😄 

I've also removed the usage of `this.` since it's not necessary, and not something we do in the rest of the codebase 😄, I've also inverted the if statements, and returned early instead, this reduces the amount of nesting, which makes the code a bit more legible. 

Lastly I've used the `TryGetValue` method instead of `ContainsKey` since this again simplifies it a bit, and IIRC there's a minor performance benefit to it 😄